crates/llm-ls/src/adaptors.rs

```diff
@@ -0,0 +1,163 @@
+use super::{
+    internal_error, APIError, APIResponse, CompletionParams, Generation, Ide, NAME, VERSION,
+};
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, USER_AGENT};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tower_lsp::jsonrpc;
+
+fn build_tgi_body(prompt: String, params: CompletionParams) -> Value {
```

Comment:
```suggestion
fn build_tgi_body(prompt: String, params: RequestParams) -> Value {
```
and clone in the caller